### ISSUE TITLE
Extending a Python class

### ISSUE THREAD

**Issue by niutech** [stevengj/PyCall.jl#138]
How to convert the following Python code:
```Python
class MyWindow(Gtk.Window):

    def __init__(self):
        Gtk.Window.__init__(self, title="Hello World")
        self.button = Gtk.Button(label="Click Here")
        self.button.connect("clicked", self.on_button_clicked)
        self.add(self.button)

    def on_button_clicked(self, widget):
        print("Hello World")

win = MyWindow()
```
to Julia for use in GTK3 app? Thanks a million!

**Comment by stevengj** [comment]
Right now there is no easy way to extend a Python class.  It's possible, but it's rather low-level, nearly at the level of the C API.  See, for example, how PyCall creates a new IO class in `src/io.jl`.

Making a higher-level interface should be very feasible and would be useful.   The key question is to decide what a higher-level Julia interface would look like, and this isn't clear to me.

**Comment by niutech** [comment]
Thanks for your quick answer. What I want to achieve is to build a GTK+ app in Julia using PyGObject (Gtk.jl is a no-go). Showing the simpliest window works OK, but I don't know how to write the event handlers in Julia and connect them to GTK+ signals. The sample code above comes from the [Python GTK+ 3 Tutorial](http://python-gtk-3-tutorial.readthedocs.org/en/latest/introduction.html#extended-example) and it seems like a good starting point, but it involves inheritance.

**Comment by PallHaraldsson** [comment]
Is this same as https://github.com/stevengj/PyCall.jl/pull/250 and should be closed? Maybe not the only overlooked issue that could be closed?

**Comment by stevengj** [comment]
Yes, I think this was closed by #250.
